### ISSUE TITLE
Gradle build error

### ISSUE THREAD

**Issue by anup-geeky** [singular-labs/Singular-Flutter-SDK#16]
I am integrating flutter plugin. When I'm adding 

  implementation 'com.google.android.gms:play-services:6.5.87'
  implementation fileTree(dir: 'libs', include: ['*.jar'])
  implementation 'com.android.support:appcompat-v7:21.0.3'
  
  I'm getting multiple duplicate class build errors.
  
duplicate class com.google.firebase.analytics.firebaseanalytics$userproperty found in modules jetified-firebase-analytics-impl-12.0.1-runtime (com.google.firebase:firebase-analytics-impl:12.0.1) and jetified-play-services-measurement-api-20.0.2-runtime (com.google.android.gms:play-services-measurement-api:20.0.2)

my build.gradle is
    implementation 'androidx.multidex:multidex:2.0.1'
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation platform('com.google.firebase:firebase-bom:29.0.0')
    implementation 'com.google.firebase:firebase-crashlytics-ktx'
    implementation 'com.google.firebase:firebase-analytics-ktx'
    //implementation 'com.github.a914-gowtham:android-video-trimmer:1.7.0'
    //implementation 'com.github.a914-gowtham:android-video-trimmer-litr:1.5.11'
    //implementation 'com.karumi:dexter:6.2.2'
    implementation 'androidx.appcompat:appcompat:1.2.0'
    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
    implementation 'com.github.bumptech.glide:glide:3.7.0'
    implementation 'io.reactivex.rxjava2:rxandroid:2.0.1'
    implementation 'io.reactivex.rxjava2:rxjava:2.1.5'
    implementation 'com.tbruyelle.rxpermissions2:rxpermissions:0.9.4@aar'
    implementation 'com.jakewharton:butterknife:10.0.0'
    implementation 'androidx.recyclerview:recyclerview:1.1.0'
    annotationProcessor 'com.jakewharton:butterknife-compiler:10.0.0'
    //compile files('libs/aspectjrt-1.7.3.jar')
    //compile files('libs/isoparser-1.0.6.jar')
    implementation ('com.googlecode.mp4parser:isoparser:1.0.6'){
        exclude  group: 'com.googlecode.mp4parser', module: 'isoparser'}
    implementation 'org.aspectj:aspectjrt:1.8.2'
    implementation('com.github.AbedElazizShe:LightCompressor:1.0.0')
    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1"
    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1"
    implementation 'com.google.android.gms:play-services-tagmanager:18.0.1'
    implementation "com.truecaller.android.sdk:truecaller-sdk:2.6.0"
    //implementation 'com.facebook.android:facebook-applinks:14.0.0'
    implementation 'com.facebook.android:facebook-android-sdk:12.3.0'
    implementation 'com.google.firebase:firebase-messaging:23.0.6'
    implementation 'com.google.code.gson:gson:2.9.0'
    implementation 'com.parse.bolts:bolts-tasks:1.4.0'
    implementation 'com.parse.bolts:bolts-applinks:1.4.0'
    implementation 'com.android.installreferrer:installreferrer:2.2'
    implementation 'com.google.firebase:firebase-dynamic-links-ktx'

How to fix it.

**Comment by MaxEliezerSingular** [comment]
Thanks for the report, and apologies for the very long delay in responding.

**This isn't a defect in the Singular plugin** — it's a dependency conflict introduced by the artifacts added alongside it.
The duplicate class you're seeing comes from two generations of Firebase Analytics landing on the same classpath:

com.google.android.gms:play-services:6.5.87 is the old monolithic Play Services artifact (2014). It pulls in every Play Services module, including the legacy firebase-analytics-impl.

firebase-bom:29.0.0 / firebase-analytics-ktx resolve to the modern play-services-measurement-api, which contains the same classes under the current packaging.

Hence firebaseanalytics$userproperty being found in both firebase-analytics-impl:12.0.1 and play-services-measurement-api:20.0.2.

Separately, com.android.support:appcompat-v7:21.0.3 is the pre-AndroidX support library, and mixing it with androidx.appcompat:1.2.0 in one build causes its own duplicate-class failures.

**To resolve:**

Remove implementation 'com.google.android.gms:play-services:6.5.87'. Never depend on the monolithic artifact — declare only the specific modules you need (e.g. play-services-ads, play-services-location).

Remove com.android.support:appcompat-v7 and com.android.support.constraint:constraint-layout, and use their AndroidX equivalents (androidx.appcompat:appcompat, androidx.constraintlayout:constraintlayout). Ensure android.useAndroidX=true and android.enableJetifier=true are set in gradle.properties.

Let the Firebase BOM manage Analytics — don't pin firebase-analytics-impl or any play-services-measurement-* version directly.

The Singular Android SDK declares its own Play Services and install-referrer dependencies at compatible versions, so it doesn't need any manual GMS artifact added on your behalf.

Closing this as it's a build configuration issue rather than a plugin bug, and the report is some time old. Please open a new issue if you're still blocked on a current version.

Thanks,
Eliezer.
